src/components/periodic-report/handlers/sync-progress-report-to-engagement.handler.ts

```diff
@@ -26,10 +32,14 @@ export class SyncProgressReportToEngagementDateRange
   constructor(
     private readonly periodicReports: PeriodicReportService,
     private readonly engagements: EngagementService,
+    private readonly config: ConfigService,
     @Logger('progress-report:engagement-sync') private readonly logger: ILogger
   ) {}
 
   async handle(event: SubscribedEvent) {
+    if (this.config.periodicReportMigration) {
```

Comment:
During migration, we delete engagement overrides if they are incorrect since we have bad data from V2. This is to prevent unwanted effects during the migration when we update the engagement. This will also prevent periodic reports from being generated when the project dates are added as well, so we need to remove it when the periodic report migration is done. Perhaps there's a better way to accomplish this.